### ISSUE TITLE
Server crash if opened with Firefox (mobil)

### ISSUE THREAD

**Issue by Blackdead** [jfrimmel/FriServer#2]
The server crashes if you try to open a secured page with Firefox Mobil.
```
Traceback (most recent call last): 
 File "SecuredFriServer.py", line 32, in _open_new_co
nnection 
   connection = self._context.wrap_socket(connection, server_side = True) 
 File "/usr/lib/python3.6/ssl.py", line 407, in wrap_socket 
   _context=self, _session=session) 
 File "/usr/lib/python3.6/ssl.py", line 814, in _init_ 
   self.do_handshake() 
 File "/usr/lib/python3.6/ssl.py", line 1068, in do_handshake 
   self._sslobj.do_handshake() 
 File "/usr/lib/python3.6/ssl.py", line 689, in do_handshake 
   self._sslobj.do_handshake() 
ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknow
n (_ssl.c:777) 

During handling of the above exception, another exception occurred: 

Traceback (most recent call last): 
 File "./main.py", line 25, in <module> 
   server.serve() 
 File "FriServer.py", line 39, in serve 
   self._open_new_connection(sock, inputs) 
 File "SecuredFriServer.py", line 37, in _open_new_co
nnection 
   self._redirect_to_https(http_socket) 
 File "SecuredFriServer.py", line 42, in _redirect_to
_https 
   data = http_socket.recv(1024) 
ConnectionResetError: [Errno 104] Connection reset by peer

```

**Comment by jfrimmel** [comment]
This should be fixed with the commit `b223a80abdf858e904f34f72e6fae6196d3ac354`.
Further testing is required.

**Comment by Blackdead** [comment]
Seem to work fine.
